textile/chat-features.textile

```diff
@@ -191,7 +191,8 @@ As well as user-initiated operations, the room must monitor its underlying resou
 *** @(CHA-RL4a3)@ @[Testable]@ If a room lifecycle operation is in progress, then a pending discontinuity event shall be recorded for this contributor - though it must not overwrite any existing discontinuity event. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change. The event will be notified to the contributor at a later point, as noted in this specification.
 *** @(CHA-RL4a4)@ @[Testable]@ If a room lifecycle operation is not in progress, then a discontinuity event will immediately be emitted to the contributor. The @ErrorInfo@ associated with the discontinuity shall be the @reason@ for the underlying channel state change.
 ** @(CHA-RL4b)@ The state monitor must handle non-@UPDATE@ channel state events.
-*** @(CHA-RL4b1)@ @[Testable]@ If a room lifecycle operation is in progress, and the new channel state is @ATTACHED@, and the @resumed@ flag is false, @and@ the particular contributor has been attached previously (i.e. a previous call to @attach()@ on the channel, per @CHA-RL1f@, has succeeded), then a pending discontinuity event will be recorded for the contributor. The error associated with this event shall be the @reason@ for the channel state change.
+*** @(CHA-RL4b1)@ This clause has been replaced by "@CHA-RL4b11@":#CHA-RL4b11.
+*** @(CHA-RL4b11)@ @[Testable]@ If a room lifecycle operation is in progress, and the new channel state is @ATTACHED@, and the @resumed@ flag is false, @and@ the particular contributor has been attached previously (i.e. a previous call to @attach()@ on the channel, per @CHA-RL1f@, has succeeded), then a pending discontinuity event will be recorded for the contributor - though it must not overwrite any existing discontinuity event. The error associated with this event shall be the @reason@ for the channel state change.
```

Comment:
I also strongly believe we should avoid exposing too much internal information unless absolutely necessary. Keeping things simple will help customers get the SDK up and running with minimal setup, without introducing unnecessary complexities that could make it harder to understand.

Reply:
The intention always was that there would be at most one discontinuity event for any given period of discontinuity (i.e. period of the room status not being ATTACHED). The reason for this was so that we don't spam users: ultimately, they need to know that there was a discontinuity (and ideally the reason for it), they don't need to know a play-by-play of all the discontinuities that may occur as part of a retry loop.

So the spec should reflect the idea of "the first discontinuity observed".

Whilst I agree that other features are not as important in terms of discontinuity, we wanted to add the feature so that it was consistent across chat features.

Reply:
> Like [current chat-js impl](https://github.com/ably/ably-chat-js/blob/58a4fa2821e098c7a09be222b979cae869729909/src/core/room-lifecycle-manager.ts#L100). stores "the latest discontinuity observed", this also means we don't need a queue?

chat-js stored the first one observed, whenever a discontinuity trigger comes up it checks if ones already stored first

Reply:
The second point seems to be related to https://github.com/ably/specification/issues/239.
So, does it really make sense to only store `first` discontinuity observed? Maybe they wouldn't care much about error caused by [RTL2f](https://sdk.ably.com/builds/ably/specification/main/features/#RTL2f), but if app has some restrictions over publish/subscribe rate, then they would like to know and fix it in the long term.

Reply:
Like the criteria can be, discontinuity `error` can be overridden if the previous+current state is `ATTACHED` and `resumed` is false over the same connection.

Reply:
Closing this, since we agree to store only the original discontinuity.